### PR TITLE
feat(api): new endpoint PartialUpdateOpenIdConfiguration

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New endpoint PartialUpdateOpenIdConfiguration, mostly based on UpdateOpenIdConfiguration
```
PATCH /administration/authentication/providers/openid
```

Fix in UpdateOpenIdConfiguration: removed unecessary double transaction

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

### CODE REVIEW

File: centreon/src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/PartialUpdateOpenIdConfiguration/PartialUpdateOpenIdConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+                    $customConfig->getAuthenticationConditions()
+                );
+            } elseif ($paramName === 'groups_mapping') {
+                $requestArray['groups_mapping'] = $this->createUpdatedGroupsMapping(
+                    $paramValue,
+                    $customConfig->getGroupsMapping()
+                );
+            } elseif ($paramValue instanceOf NoValue) {
+                $requestArray[$paramName] = $oldConfigArray[$paramName];
+            }
```
Comment (jeremyjaouen) [inline centreon/src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/PartialUpdateOpenIdConfiguration/PartialUpdateOpenIdConfiguration.php:167]: You can use a switch to be more concise

File: centreon/src/Core/Security/ProviderConfiguration/Domain/OpenId/Model/CustomConfiguration.php [hunk 2]
```diff
@@ -33,7 +33,7 @@
 use Core\Security\ProviderConfiguration\Domain\Model\GroupsMapping;
 use TypeError;
 
-final class CustomConfiguration implements CustomConfigurationInterface, OpenIdCustomConfigurationInterface
```
Comment (callapa) [inline centreon/src/Core/Security/ProviderConfiguration/Domain/OpenId/Model/CustomConfiguration.php:36]: Why remove 'final' keyword ?

Comment (a-launois) [reply centreon/src/Core/Security/ProviderConfiguration/Domain/OpenId/Model/CustomConfiguration.php:36]: I need to be able to mock the class in the tests

File: centreon/src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/PartialUpdateOpenIdConfiguration/PartialUpdateOpenIdConfiguration.php [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    private function createUpdatedCustomConfiguration(
+        CustomConfigurationInterface $customConfig,
+        PartialUpdateOpenIdConfigurationRequest $request
+    ): CustomConfiguration {
+        /** @var CustomConfiguration $customConfig */
+        $requestArray = $request->toArray();
+        $oldConfigArray = $customConfig->toArray();
+
+        foreach ($requestArray as $paramName => $paramValue) {
+            if ($paramName === 'contact_template') {
```
Comment (adr-mo) [inline centreon/src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/PartialUpdateOpenIdConfiguration/PartialUpdateOpenIdConfiguration.php:141]: Could it be clearer and cleaner to use a switch case ?

File: centreon/src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/PartialUpdateOpenIdConfiguration/PartialUpdateOpenIdConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+     */
+    private function findAccessGroupFromFoundAccessGroups(
+        int $accessGroupIdFromRequest,
+        array $foundAccessGroups
+    ): ?AccessGroup {
+        foreach ($foundAccessGroups as $foundAccessGroup) {
+            if ($accessGroupIdFromRequest === $foundAccessGroup->getId()) {
+                return $foundAccessGroup;
+            }
+        }
```
Comment (adr-mo) [inline centreon/src/Core/Security/ProviderConfiguration/Application/OpenId/UseCase/PartialUpdateOpenIdConfiguration/PartialUpdateOpenIdConfiguration.php:305]: Could be done using `array_filter`

**Review by adr-mo — APPROVED** [review]

**Pull request merged**